src/Castle.Windsor/MicroKernel/Registration/BasedOnDescriptor.cs

```diff
@@ -535,11 +536,10 @@ private static bool IsBasedOnGenericInterface(Type type, Type basedOn, out Type[
 			var types = new List<Type>(4);
 			foreach (var @interface in type.GetInterfaces())
 			{
-				if (@interface.IsGenericType &&
+				if (@interface.GetTypeInfo().IsGenericType &&
 				    @interface.GetGenericTypeDefinition() == basedOn)
 				{
-					if (@interface.ReflectedType == null &&
-					    @interface.ContainsGenericParameters)
+					if (@interface.GetTypeInfo().ContainsGenericParameters)
 					{
```

Comment:
Should we wrap ReflectedType in a #if? It is in net45, and also added in netstandard2.0.